### PR TITLE
Generic recorder

### PR DESCRIPTION
**Pull request by sjoeman1**
Made the Recorder class more generic. This means we can use the same class for recording the the blackbox inputs

### CODE REVIEW

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp [hunk 1]
```diff
@@ -1,36 +1,40 @@
...
 	}
 	
 	// create and open new recording file with current timestamp
 	std::time_t t = std::time(nullptr);
 	std::tm tm = *std::localtime(&t);
 	std::stringstream buffer;
 	buffer << std::put_time(&tm, "%Y%m%d-%H%M%S");
-	std::string fileName = dirName + "\\Record" + buffer.str() + ".txt";
-	m_recordingFile.open(fileName, std::ios::binary | std::ios::app);
+	std::string path = dirPath + "\\" + p_fileName + buffer.str() + ".txt";
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp:32]: Not a huge problem but a suggestion, you could have a parameter that uses the [strftime](https://en.cppreference.com/w/c/chrono/strftime) format so the creator can decide on the file format.

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp [hunk 2]
```diff
@@ -48,15 +52,15 @@ Recorder::~Recorder()
...
 /// @author Sjoerd Schilder
 ///
-void Recorder::WriteRecording(float* p_input, double currentTime)
+void Recorder::WriteRecording(const float* p_input, const double p_currentTime, const bool p_compression)
 {
-	// doesnt write if the input is the same as the previous time
-	if (CheckSameInput(p_input)) return;
-	m_recordingFile << currentTime << " ";
-	for (int i = 0; i < PARAMETERS; i++)
+	// doesn't write if the input is the same as the previous time
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp:60]: Comment should be updated now that the behaviour has (slightly) changed

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp (hunk 1 above)
Comment (sjoeman1) [reply speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp:32]: i don't think that isreally necesarry , maybe i something i could look at later

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp [hunk 3]
```diff
@@ -69,16 +75,14 @@ void Recorder::WriteRecording(float* p_input, double currentTime)
 ///
 /// @brief Checks if the input is the same as the previous input
 /// @preconditions p_input must be PARAMETERS long (in this case 4)
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp:77]:
```suggestion
/// @preconditions p_input must be m_paramAmount long (in this case 4)
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp [hunk 4]
```diff
@@ -69,16 +75,14 @@ void Recorder::WriteRecording(float* p_input, double currentTime)
...
-/// @param isSame boolean ref that becomes false if the input is differrent
 /// @param p_input array of size PARAMETERS
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp:78]:
```suggestion
/// @param p_input array of size m_paramAmount
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp [hunk 5]
```diff
@@ -48,15 +52,17 @@ Recorder::~Recorder()
 /// Only writes to the file if the input is different then the previous input.
 /// @preconditions p_input must be PARAMETERS long (in this case 4)
 /// @param p_input array of size PARAMETERS
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp:54]: Also please edit the start of the comment `float*` to `array`

```suggestion
/// @preconditions p_input must be m_paramAmount long (in this case 4)
/// @param p_input array of size m_paramAmount 
```

**Review by larsdekwant — CHANGES_REQUESTED** [review]

**Review by larsdekwant — CHANGES_REQUESTED** [review]

**Review by larsdekwant — APPROVED** [review]

**Review by suppergerrie2 — APPROVED** [review]

**Pull request merged**